### PR TITLE
4733 Add random sorting on titles

### PR DESCRIPTION
**Pull request by RemcoSimonides**
Fix issue #4733 for movie titles

### CODE REVIEW

File: apps/festival/festival/src/app/marketplace/home/home.component.html [hunk 1]
```diff
@@ -47,7 +47,7 @@ <h2>{{ section.title }}</h2>
       </a>
     </header>
     <bf-carousel flex>
-      <ng-container *ngFor="let movie of section | queryTitles | async">
+      <ng-container *ngFor="let movie of section | queryTitles | async | sortArrayBy:section.sorting">
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/marketplace/home/home.component.html:50]: Rename `sortArrayBy` into `sortBy`. We can only sort list anyway

File: libs/utils/src/lib/pipes/sort-array.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export type SortingOptions = 'default' | 'random';
+export const sortingOptions: SortingOptions[] = ['default', 'random'];
+
+@Pipe({ name: 'sortArrayBy' })
+export class SortArrayByPipe implements PipeTransform {
+  transform(array: any[], method: SortingOptions) {
+    if (Array.isArray(array)) {
+      if (method === 'random') {
+        for (let i = array.length - 1; i > 0; i--) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pipes/sort-array.pipe.ts:11]: Use the `array.sort((a, b) => )` method : 
```typescript
const sortFn = {
  random: (a, b) => Math.random() - Math.random() // or  Math.random() - 0.5
}
export type SortingOption = 'default' | keyof typeof sortFn;

@Pipe({ name: 'sortBy' })
export class SortByPipe implements PipeTransform {
  transform(array: any[], method: SortingOptions = 'default') {
    if (method === 'default' || !Array.isArray(array)) return array;
    return array.sort(sortFn[method]);
  }
}
```

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/utils/src/lib/pipes/sort-array.pipe.ts (hunk 2 above)
Comment (RemcoSimonides) [reply libs/utils/src/lib/pipes/sort-array.pipe.ts:11]: image

Comment (pldespaigne) [reply libs/utils/src/lib/pipes/sort-array.pipe.ts]: Hum interesting 🤔 
More details in the "solution" of [this page](https://javascript.info/task/shuffle)

TL;DR :
`.sort(() => .5 Math.random())` is bad if you use it on large inputs or if you want to have an evenly distributed results over many iterations.

IMO our use case is simple enough to use `.sort(() => .5 Math.random())`, moreover because performance doesn't seems to be an issue in this case, I think we should favor simplicity of code.

**Review by GrandSchtroumpf — APPROVED** [review]

File: libs/utils/src/lib/pipes/sort-array.pipe.ts (hunk 2 above)
Comment (GrandSchtroumpf) [reply libs/utils/src/lib/pipes/sort-array.pipe.ts]: lol. yes we just want an insignificant shuffle.
